### PR TITLE
test(frontend): transactions store

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Before breaking everything, here's a bare minimal test for the `transactions.store`.

### CODE REVIEW

File: src/frontend/src/tests/lib/stores/transactions.store.spec.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+				store.subscribe((state) => {
+					expect(state?.[tokenId]).toHaveLength(2);
+
+					done();
+				})();
+			}));
+	});
+
+	describe('append', () => {
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/tests/lib/stores/transactions.store.spec.ts:64]: maybe here the same test to check if no duplicates? as the one for prepend?

File: src/frontend/src/tests/lib/stores/transactions.store.spec.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+				const store = initTransactionsStore<IcTransactionUi>();
+
+				const transactions = [
+					createCertifiedTransaction('tx1'),
+					createCertifiedTransaction('tx2'),
+					createCertifiedTransaction('tx3')
+				];
+
+				store.append({ tokenId, transactions });
+				store.cleanUp({ tokenId, transactionIds: ['tx1', 'tx3'] });
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/tests/lib/stores/transactions.store.spec.ts:97]: if you add `tx4` just to check the behavior for non existing transactions, does the test fails? or is it no issue? I don't remember the code

**Review by AntonioVentilii-DFINITY — APPROVED** [review]
Already good, tks!

A few suggestions for further tests, but only if you think necessary

File: src/frontend/src/tests/lib/stores/transactions.store.spec.ts (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/tests/lib/stores/transactions.store.spec.ts:64]: that's actually a good point, turns out `append` do not deduplicate. Should we improve the store in another PR and provide the test at the same time?

File: src/frontend/src/tests/lib/stores/transactions.store.spec.ts (hunk 2 above)
Comment (peterpeterparker) [reply src/frontend/src/tests/lib/stores/transactions.store.spec.ts:97]: It just do nothing. Added a test 632a63dfa38f32a61ee2290f47a43ba19bdbfe19.

File: src/frontend/src/tests/lib/stores/transactions.store.spec.ts (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/tests/lib/stores/transactions.store.spec.ts:64]: Yes! I didn't even know that it wasn't doing it, good thing you tested!

**Pull request merged**